rats-devtools/src/python/rats/docs/_commands.py

```diff
@@ -58,171 +98,182 @@ def build_tutorial_notebooks() -> None:
-                rmtree(notebooks_target_path, ignore_errors=True)
-                notebooks_target_path.mkdir(parents=True, exist_ok=True)
-                source_file_names = [f.stem for f in jupytext_sources_path.glob("*.py")]
-
-                # symlink the jupytext files to the target path
-                for file_name in source_file_names:
-                    (jupytext_sources_path / f"{file_name}.py").link_to(
-                        notebooks_target_path / f"{file_name}.py"
-                    )
-
-                for cmd in commands:
-                    try:
-                        subprocess.run(cmd, cwd=notebooks_target_path, check=True)
-                    except subprocess.CalledProcessError as e:
-                        sys.exit(e.returncode)
-
-                for file_name in source_file_names:
-                    # delete the .py symlink
-                    (notebooks_target_path / f"{file_name}.py").unlink()
-                    # delete the .ipynb file
-                    (notebooks_target_path / f"{file_name}.ipynb").unlink()
-                    # add the notebook to the .pages file
-
-                # add a readmd.md file to the target path saying this folder is autogenerated
-                source_rel_to_target = "../_tutorial_notebook_sources"
-                with (notebooks_target_path / "readme.md").open("w") as f:
-                    f.write(
-                        f"This folder is autogenerated from the files in {source_rel_to_target} "
-                        + "using\n```bash\nrats-devtools.pipx build-tutorial-notebooks\n```\n"
-                    )
+            _build_tutorial_notebooks()
 
         return build_tutorial_notebooks
 
-    def _do_mkdocs_things(self, cmd: str) -> None:
-        devtools_path = Path("rats-devtools").resolve()
-        root_docs_path = devtools_path / "src/resources/root-docs"
-        mkdocs_config = devtools_path / "mkdocs.yaml"
-        mkdocs_staging_path = devtools_path / "dist/docs"
-        site_dir_path = devtools_path / "dist/site"
-        mkdocs_staging_config = devtools_path / "dist/mkdocs.yaml"
-        # clear any stale state
-        rmtree(mkdocs_staging_path, ignore_errors=True)
-
-        # start with the contents of our root-docs
-        copytree(root_docs_path, mkdocs_staging_path)
-
-        components = [
-            "rats-apps",
-            "rats-devtools",
-            "rats-pipelines",
-            "rats-processors",
-        ]
 
-        for c in components:
-            component_docs_path = Path(c).resolve() / "docs"
-            symlink(component_docs_path, mkdocs_staging_path / c)
-
-        # replace the mkdocs config with a fresh version
-        mkdocs_staging_config.unlink(missing_ok=True)
-        copy(mkdocs_config, mkdocs_staging_config)
-
-        args = [
-            "--config-file",
-            str(mkdocs_staging_config),
+def _do_mkdocs_things(cmd: str) -> None:
```

Comment:
i'd like to keep these in the class so we can slowly move the side effects out to other services

Reply:
Fixed in [e6e0781](https://github.com/microsoft/rats/pull/161/commits/e6e07815cdc3c0d9b7dd0c9cc30fe917c4b2259f)